backend/tet/events/utils.py

```diff
@@ -0,0 +1,48 @@
+# Common utils used by services and transformations
+
+from django.conf import settings
+from django.http import HttpRequest
+from requests import RequestException
+from rest_framework.exceptions import PermissionDenied
+from shared.oidc.utils import get_organization_roles
+
+# TET service uses Linked Events `provider` field to store data that needs to be searchable
+# We can then filter events with the `text` criterion to reduced matches
+PROVIDER_NAME_FIELD = "fi"
+PROVIDER_BUSINESS_ID_FIELD = "sv"
+
+# AD users get this as the company name
+# while for suomi.fi users the company name is fetched via YTJ
+# TODO we need to localize this based on user's locale
+CITY_OF_HELSINKI = "Helsingin kaupunki"
+
+
+def get_business_id(request: HttpRequest):
+    if settings.NEXT_PUBLIC_MOCK_FLAG:
+        return "123456-7"
+
+    try:
+        organization_roles = get_organization_roles(request)
+    except RequestException:
+        raise PermissionDenied(
+            detail="Unable to fetch organization roles from eauthorizations API"
+        )
```

Comment:
This handles all RequestException exceptions as PermissionDenied here.
More finegrained error handling could be beneficial but I leave it up to you ponder and decide.

Reply:
Good question how the error should be handled. User needs to be linked to a company and this error means we cannot find that company, which makes the TET service unusable. So maybe a 5xx error would be more suitable?